File: app/ClinicalScheduler.API/ClinicalScheduler.Domain/Entities/Shift.cs
using ClinicalScheduler.Domain.Enums;

namespace ClinicalScheduler.Domain.Entities;

public class Shift
{
    public int Id { get; set; }
    public int StaffId { get; set; }
    public int DepartmentId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public ShiftType ShiftType { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Staff Staff { get; set; } = null!;
    public Department Department { get; set; } = null!;
}

File: app/clinical-scheduler-ui/src/types/shift.ts
export type ShiftType = "Morning" | "Afternoon" | "Night";
export type SwapStatus =
  | "PendingRequestee"
  | "PendingAdmin"
  | "Approved"
  | "Rejected"
  | "Cancelled";

export interface Shift {
  id: number;
  staffId: number;
  departmentId: number;
  staffName: string;
  staffInitials: string;
  departmentName: string;
  startTime: string;
  endTime: string;
  shiftType: ShiftType;
  notes?: string;
}

export interface SwapAuditEntry {
  at: string;
  by: string;
  action: string;
  note?: string;
}

export interface ShiftSwapRequest {
  id: number;
  requesterId: number;
  requesterName: string;
  requesterInitials: string;
  requesteeId: number;
  requesteeName: string;
  requesteeInitials: string;
  requesterShift: Shift;
  requesteeShift: Shift;
  reason: string;
  status: SwapStatus;
  submittedAt: string;
  auditEntries: SwapAuditEntry[];
}
